### PR TITLE
du error output should match GNU

### LINKED ISSUE #1770: du should match GNU's output when not enough permissions
with rm from GNU:
```
$ mkdir a
$ touch a/b
$ chmod u=rw a 
$ du a
du: cannot access 'a/b': Permission denied
```
With the Rust implementation:
```
$ mkdir a
$ touch a/b
$ chmod u=rw a 
$ du a
du: error: Permission denied (os error 13)
```

This will probably requires to update the `show_error` for all binaries (which is fine)
(might conflict with https://github.com/uutils/coreutils/issues/1769)
but the filename need to be displayed too here.

### PR DESCRIPTION
**Pull request by growse**
Fixes #1770

* Created a new error macro which allows the customization of the
  "error:" string part
* Match the du output based on the type of error encountered. Can extend
  to handling other errors I guess.

I'm pretty new to rust, so interested in feedback on this. I did contemplate changing the signature of the `show_error` macro, but it seemed less complex to create a new macro allowing the customization of the `error:` string part. Also, I'm not wild about hardcoding the `Permission denied` string, but it seemed the simplest thing to do.

### CODE REVIEW

**Comment by growse** [comment]
> Could you please add a test to make sure that the error message is correctly set and that we won't regress in the future?
> many thanks

I'll have a go!

**Comment by sylvestre** [comment]
See this PR for a similar test https://github.com/uutils/coreutils/pull/1780/files :)

**Comment by growse** [comment]
> See this PR for a similar test https://github.com/uutils/coreutils/pull/1780/files :)

I see tests in both `du.rs` and also `test_du.rs`. Interestingly the ones in `du.rs` don't get run if I `cargo test --no-default-features --features du`.

Any reason for the split, and preferences over where this test should live?

**Comment by sylvestre** [comment]
Yeah. du.rs is a unit test (ie test the rust functions), test_du.rs is an integration test (test the binary).
Both are great!

File: src/uu/du/src/du.rs [hunk 1]
```diff
@@ -175,7 +176,21 @@ fn du(
                             }
                         }
                     }
-                    Err(error) => show_error!("{}", error),
+                    Err(error) => match error.kind() {
+                        ErrorKind::PermissionDenied => {
```
Comment (sylvestre) [inline src/uu/du/src/du.rs:180]: I don't have a better proposition for now (sorry!) but this is a lot of code just to show an error  :/

Comment (growse) [reply src/uu/du/src/du.rs:180]: Errors are hard though :(

**Comment by growse** [comment]
Mmm. This might be more challenging than I thought. Turns out there's no easy cross-platform way to set a file as non-readable, so might need separate test cases for POSIX and Windows.

**Comment by sylvestre** [comment]
@growse do you think you will try to finish it?

**Comment by growse** [comment]
I will have a go. I'm just slightly weighed down under parenting at the moment, and had expected it to be a slightly easier job with testing not requiring platform-specific code.

Let me see if I can make some progress this week and try and wrap it up.

**Comment by growse** [comment]
I've got a test for non-windows cases. Windows is a little trickier though, as there's no way that I can see in the rust API to make a file unreadable. For the POSIX case, I just `chmod -r`'d one of the fixtures. On Windows, I think we'd need a call to [SetFileSecurityW](https://microsoft.github.io/windows-docs-rs/doc/bindings/Windows/Win32/Security/fn.SetFileSecurityW.html) or something more complex.

**Pull request merged**